velox/functions/prestosql/window/AggregateWindow.cpp

```diff
@@ -54,6 +55,9 @@ class AggregateWindowFunction : public exec::WindowFunction {
     // the function value for each row.
     aggregate_ = exec::Aggregate::create(
         name, core::AggregationNode::Step::kSingle, argTypes_, resultType);
+    // Set HashStringAllocator needed for out of line allocations.
+    stringAllocator_ = std::make_unique<HashStringAllocator>(mappedMemory_);
```

Comment:
Any particular reason to create new HashStringAllocator for each function? HashAggregation operator creates one HashStringAllocator and passes it to all the aggregate functions.

Reply:
I was passing the MappedMemory to all window functions for generality. There are other options we can consider :
i) We need HashStringAllocator only for Aggregate window functions. All other window functions don't need it. Do we use MappedMemory for any other purposes ? If not, then we can change this API to take a HashStringAllocator instead and that could be common for all functions in the Window operator.
ii) We could keep static variables for MappedMemory and HashStringAllocator in the AggregateWindowFunction. So then we could reuse these across aggregates. Of course we will have to change them with each operator. For that we could compare that if the present MappedMemory is different from previous one then to change it. But then these variables would need to be thread-safe also. 
So in general I seem to favor i).
wdyt ?

Reply:
> we can change this API to take a HashStringAllocator instead

Sounds good.